src/DGtal/geometry/surfaces/DigitalSurfaceConvolver.ih

```diff
@@ -1096,15 +1096,18 @@ DGtal::DigitalSurfaceConvolver< Functor, KernelFunctor, KSpace, DigitalKernel, 2
               {
                 for( KernelConstIterator itm = myKernelMask->first; itm != myKernelMask->second; ++itm )
                   {
-                    myKSpace.sSetKCoords( shiftedSpel, myKSpace.sKCoords( myKSpace.sSpel( *itm )) + shiftInnerSpel );
+                    if( myKSpace.cIsInside( myKSpace.sKCoords( myKSpace.sSpel( *itm )) + shiftInnerSpel ))
+                    {
+                      myKSpace.sSetKCoords( shiftedSpel, myKSpace.sKCoords( myKSpace.sSpel( *itm )) + shiftInnerSpel );
```

Comment:
Thanks for fix ! 
- ~~~About the direct incident request, line 1037, is there any guarantee that it does not go outside the space ? Especially for open spaces ?~~~ Ok, we are inside the shape.
- Another remark: line 1061, unfortunately, this will not work along periodic dimension but it will maybe only affect the result reuse ?